pcl_utilities/src/passthrough_filter_service.cpp

```diff
@@ -0,0 +1,203 @@
+/*
+ * Author: Brian Flynn
+ * Date: Nov 17, 2022
+ * Editors: Christian Tagliamonte
+ * Last Modified: Aug 19, 2024
+ * Adapted from:
+ * https://github.com/uml-robotics/armada_behaviors/commits/main/armada_flexbe_utilities/src/service/pcl_passthrough_filter_service.cpp
+ *
+ * Description: Starts up a service for filtering point cloud data with a bounding box.
+ *
+ * Input: sensor_msgs/PointCloud2
+ * Output: sensor_msgs/PointCloud2
+ *
+ * Usage:
+ *    `ros2 launch pcl_utilities passthrough_filter.xml`
+ */
+#include <functional>  // std::bind
+#include <limits>  // std::numeric_limits
+#include <memory>  // std::make_shared
+#include <sstream>  // std::stringstream
+#include <string_view>  // std::string_view
+#include <utility>  // std::move
+
+#include "pcl/impl/point_types.hpp"
+#include "pcl/point_cloud.h"
+#include "pcl/filters/passthrough.h"
+#include "pcl_conversions/pcl_conversions.h"
+
+#include "rclcpp/node.hpp"
+#include "rclcpp/service.hpp"
+#include "rcl_interfaces/msg/floating_point_range.hpp"
+#include "rcl_interfaces/msg/parameter_descriptor.hpp"
+#include "sensor_msgs/msg/point_cloud2.hpp"
+
+#include "pcl_utility_msgs/srv/pcl_passthrough_filter.hpp"
+
+using pcl_utility_msgs::srv::PCLPassthroughFilter;
+constexpr std::string_view g_PARAM_NAMESPACE = "filters.passthrough.";
+
+class PassthroughFilterService : public rclcpp::Node
+{
+private:
+  rclcpp::Service<PCLPassthroughFilter>::SharedPtr passthrough_filter_service_;
+  rcl_interfaces::msg::ParameterDescriptor constraints_;
+  double x_min_;
```

Comment:
Why did we switch from float to double and then add in a bunch of checks to make sure the conversion happened correctly? Why can't we just use floats?